javascript/react/common-mistakes-in-react/setstate-is-async.md

```diff
@@ -0,0 +1,100 @@
+---
+author: kapnobatai136
+  
+aspects:
+
+  - workout
+
+  - deep
+
+type: normal
+
+category: tip
+
+---
+
+# setState is async
+
+---
+## Content
+
+More often than not, we are very eager with accessing a new state value after setting it. Because a new value is set on the next available render, the state might not reflect your latest update. Let's take a look at an example:
+
+```js
+handleChange = age => {
+  this.setState({ age });
+  this.props.callback(this.state.age);
+};
+```
+
+In the `handleChange` function, when we call `this.props.callback(this.state.count)` we are getting the old state value, not the new one we set through `this.setState({ count })`. This issue can be fixed by using an optional second argument of the `setState` method, a callback function that is called after the state is updated:
```

Comment:
```suggestion
In the `handleChange` function, when we call `this.props.callback(this.state.count)` we are getting the old state value, not the new one we set through `this.setState({ count })` (because the state hasn't yet updated between the two lines of code). This issue can be fixed by using an optional second argument of the `setState` method, a callback function that is called after the state is updated:
```